### PR TITLE
⚡ Add state-preserving redirect feature

### PR DESCRIPTION
**Pull request by thomasogara**
### Change List:
* Submitting a comment will no longer result in the page reloading with the default number of visible comments. Now, the page will reload with the same number of visible comments as had been set when the form was posted, with the updated comment at the top of the list.

#### Posting Comment
image
#### After Posting
image

**As can be seen from the above, the page was reloaded, but the number of comments being displayed changes on reload**

#### Posting Comment
image
#### After Posting
image

**As can be seen from the above, the page now reloads and will display the same number of comments as had been visible before posting, with the posted comment now displayed as the top comment**

### CODE REVIEW

File: portfolio/src/main/webapp/script.js [hunk 1]
```diff
@@ -154,12 +154,47 @@ const selectionChangeHandler = async (value) => {
...
+  const optionValues = [];
+  // constructing the optionValues[] array in this way is required as
+  // maxCommentsElement.options is an HTMLCollection object, and does not
+  // have a .indexOf() function
+  for (let option of maxCommentsElement.options) {
+    optionValues.push(option.value);
+  }
+  const index = optionValues.indexOf(value);
+  // if the passed value is not a valid option (not found in options array), set the value to 5
+  if (index === -1) {
```
Comment (gjrb) [inline portfolio/src/main/webapp/script.js:183]: const notFound = -1

File: portfolio/src/main/webapp/script.js [hunk 2]
```diff
@@ -154,12 +154,47 @@ const selectionChangeHandler = async (value) => {
...
+    // option with value 5 is stored at index 0
+    index = 0;
```
Comment (gjrb) [inline portfolio/src/main/webapp/script.js:185]: You set index as a const, but if you hit this condition you then change index so it's not really a constant.

File: portfolio/src/main/webapp/script.js [hunk 3]
```diff
@@ -154,12 +154,47 @@ const selectionChangeHandler = async (value) => {
...
+
+/**
+ * Set the value of the #maxComments element on the page
+ * @param{Number} value The value to be assigned
+ */
+const setMaxComments = (value) => {
+  const maxCommentsElement = document.getElementById('maxComments');
+  const selectedIndex = maxCommentsElement.selectedIndex;
+  const optionValues = [];
+  // constructing the optionValues[] array in this way is required as
```
Comment (gjrb) [inline portfolio/src/main/webapp/script.js:175]: I would put the comment above optionValues since that is part the code you are explaining

Comment (thomasogara) [reply portfolio/src/main/webapp/script.js:174]: fixed 👍 https://github.com/thomasogara/step-repository/pull/27/commits/7389f7020e377c9ab3600a40b57a37064759690e

File: portfolio/src/main/webapp/script.js (hunk 2 above)
Comment (thomasogara) [reply portfolio/src/main/webapp/script.js:187]: my mistake, fixed 👍 https://github.com/thomasogara/step-repository/pull/27/commits/c006624b00cb551b6e54d27bfa80c91199b53284

File: portfolio/src/main/webapp/script.js (hunk 1 above)
Comment (thomasogara) [reply portfolio/src/main/webapp/script.js]: implemented now 👍 https://github.com/thomasogara/step-repository/pull/27/commits/c006624b00cb551b6e54d27bfa80c91199b53284

**Pull request merged**